### PR TITLE
A browser init failure should cause a restart

### PR DESCRIPTION
**Pull request by gsnedders**
It makes no sense to call the `init` function and then not call `init_failure`, aside from the fact that it previously caused an `AttributeError` due to `browser_pid` not being initialized.

This means that if the browser has partially started (e.g., if the WebDriver server has started, but `wait_for_service` never actually connected) we actually stop it before retrying.

While we're at it, add a test for what we do with `process_output`  when `wait_for_service` never connects. (I am happy to entertain the idea that the current behaviour is wrong and change the implementation and expectation.)

### CODE REVIEW

File: tools/wptrunner/wptrunner/tests/browsers/test_base.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    logger = structuredlog.StructuredLogger("test")
+    handler = MozLogTestHandler()
+    logger.add_handler(handler)
+
+    class CustomException(Exception):
+        pass
+
+    with mock.patch.object(base, "wait_for_service", side_effect=CustomException):
+        browser = base.WebDriverBrowser(
+            logger, webdriver_binary="echo", webdriver_args=["sample output"]
```
Comment (jgraham) [inline tools/wptrunner/wptrunner/tests/browsers/test_base.py:35]: Does this depend on `echo` being a real binary that actually does what you expect?

Comment (gsnedders) [reply tools/wptrunner/wptrunner/tests/browsers/test_base.py:35]: Yes, but that should be true on any POSIX system (and is true on every vaguely POSIX system I'm aware of).

Comment (gsnedders) [reply tools/wptrunner/wptrunner/tests/browsers/test_base.py:35]: Ah, it's not true on Windows (where it's only an internal command of `cmd.exe`).

Unfortunately it's hard to partially mock `mozprocess`, so I guess let's just skip this on Windows.

File: tools/wptrunner/wptrunner/testrunner.py [hunk 2]
```diff
@@ -513,14 +514,8 @@ def init(self):
...
-            return RunnerManagerState.error()
-        elif not result:
-            return RunnerManagerState.initializing(self.state.test_type,
-                                                   self.state.test,
-                                                   self.state.test_group,
-                                                   self.state.group_metadata,
-                                                   self.state.failure_count + 1)
+        if not result:
+            return self.init_failed()
         else:
```
Comment (jgraham) [inline tools/wptrunner/wptrunner/testrunner.py:519]: FWIW you could also remove the `else ` after `return`.

**Review by jgraham — APPROVED** [review]

**Pull request merged**